SO3_XL_ToolChange_fixedLocation

```diff
@@ -1,21 +1,39 @@
-;This macro allows you to use a fixed machine location for a tool change/probe. Ideal when you're workpiece surface has been carved away.
+; This macro allows you to use a fixed machine location for a tool change and probe. It works by
+; probing twice, once to capture the position of the tool you're changing away from, and again to
+; set the adjusted WCS offset of the tool you're changing to.
+
+; Using the macro in practice:
+; 1) Insert an M6 tool change command in your program, which will pause cncjs and allow you to
+;     execute this macro.
+; 2) The router retracts to a safe height and moves to a pre-defined location for tool changes and
+;     probing, then pauses.
+; 3) Turn off your router, set up your probe, and continue. This will probe to capture the position
+;     of your current tool and then retract back up to the tool change height and pause.
+; 4) Change out your tool, set up your probe, and continue. This will probe the new tool. When it
+;     reaches the probe, the end of the tool will be in precisely the same position as the original
+;     tool, but the Z position will be different, so we set the current Z offset to be the previously
+;     captured offset.
+; 5) Remove the probe and continue. The spindle will move back to the location it encountered
+;     the M6.
+; 6) Turn your router back on and resume execution of your program.
 
-; Wait until the planner queue is empty
 %wait
-; Set user-defined variables
-%SAFE_HEIGHT = -5	;clear everything height (negative number, distance below Z limit)
-;Following set probe location
-%TOOL_PROBE_X = -30	;machine coordinates
-%TOOL_PROBE_Y = -350	;machine coordinates
-%TOOL_PROBE_Z = -20	;machine coordinates --> lower this (more negative) to start the probing closer to wasteboard
+G21 ; Metric
```

Comment:
Good catch...this should come after the section where we save the modal states to ensure that when the tool change is complete, the correct units are restored.

Reply:
So I was originally storing and restoring the coordinates in the current mode (imperial or metric), but I found that it wasn't restoring correctly. I'm honestly not sure why, and came to this final result by trial and error. This way seems to work consistently: save and restore in metric, and then switch back to the original mode.

Reply:
@zachallaun I can't replicate the errors you were seeing.